components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -392,6 +426,171 @@ public void updateLocalClaim(String claimId, LocalClaimReqDTO localClaimReqDTO)
     }
 
+    /**
+     * Updates a claim dialect with related claims from an uploaded file.
+     *
+     * @param fileInputStream   InputStream representing the uploaded claim dialect file.
+     * @param fileDetail        Attachment object with metadata about the uploaded claim dialect file.
+     * @return a String representing the updated claim dialect's resource identifier.
+     */
+    public String updateClaimDialectFromFile(InputStream fileInputStream, Attachment fileDetail) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating claim dialect from file.");
+        }
+
+        String dialectId = null;
+        try {
+            ClaimDialectConfiguration dialectConfiguration = getDialectFromFile(fileInputStream, fileDetail);
+            dialectId = dialectConfiguration.getId();
+            if (dialectId.equals(LOCAL_DIALECT_PATH)) {
+                updateLocalClaims(dialectConfiguration.getLocalClaimReqDTOList());
+            } else {
+                updateClaimDialect(dialectId, dialectConfiguration.getClaimDialectReqDTO());
+                updateExternalClaims(dialectId, dialectConfiguration.getExternalClaimReqDTOList());
+            }
+            return dialectId;
+        } catch (ClaimMetadataException e) {
+            throw handleClaimManagementException(e, ERROR_CODE_ERROR_UPDATING_DIALECT, dialectId);
+        }
+    }
+
+    private void updateLocalClaims(List<LocalClaimReqDTO> localClaimReqDTOList) throws ClaimMetadataException {
+
+        for (LocalClaimReqDTO localClaimReqDTO : localClaimReqDTOList) {
+            if (isLocalClaimExist(localClaimReqDTO.getClaimURI())) {
+                updateLocalClaim(getResourceId(localClaimReqDTO.getClaimURI()), localClaimReqDTO);
+            } else {
+                addLocalClaim(localClaimReqDTO);
+            }
+        }
+    }
+
+    private void updateExternalClaims(String dialectId, List<ExternalClaimReqDTO> externalClaimReqDTOList)
+            throws ClaimMetadataException {
+
+        for (ExternalClaimReqDTO externalClaimReqDTO : externalClaimReqDTOList) {
+            String claimId = getResourceId(externalClaimReqDTO.getClaimURI());
+            if (isExternalClaimExist(dialectId, claimId)) {
+                updateExternalClaim(dialectId, claimId, externalClaimReqDTO);
+            } else {
+                addExternalClaim(dialectId, externalClaimReqDTO);
+            }
+        }
+    }
+
+    /**
+     * Exports a claim dialect with related claims in the specified file type.
+     *
+     * @param dialectId     ID of the claim dialect to export.
+     * @param fileType      Type of file to export the claim dialect to.
+     * @return a FileContent object representing the exported claim dialect file.
+     */
+    public FileContent exportClaimDialectToFile(String dialectId, String fileType) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Exporting Claim Dialect for ID " + dialectId + " as a " + fileType + " file.");
```

Comment:
Shall we format this String using String.format

`String.format("Exporting Claim Dialect for ID %s as a %s file.", dialectId, fileType);`

Reply:
Reformatted in all places